### PR TITLE
Lvnod 65 create model event

### PR DESCRIPTION
**Pull request by yarvanya**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LVNOD-65)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-66)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-67)

## Code reviewers

- [x] @yurii-kokhalevych
- [x] @denisbodnar
- [x] @yuliannakorol

### Second Level Review

- [x] @lhalam
- [ ] @4asof4ik

## Summary of issue

First, I've created model for the Event model, then its seed and sequelize automatially generated migration for the model.
Second, I've created controller to this model with all needed methods: create,update,destroy,list and retrieve(list by id), and added routes for it.
Then I've added some change for user model and corrected some small mistakes.

## CHECK LIST
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR

### CODE REVIEW

File: server/controllers/index.js [hunk 1]
```diff
@@ -1,9 +1,11 @@
 const profiles = require('./profile.controller');
 const users = require('./user.controller');
 const autorization = require('./auth.controller');
+const eventes = require('./event.controller');
```
Comment (denisbodnar) [inline server/controllers/index.js]: events

File: server/controllers/index.js [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
 
 module.exports = {
   profiles,
   users,
   autorization,
+  eventes
```
Comment (denisbodnar) [inline server/controllers/index.js]: events

**Review by denisbodnar — DISMISSED** [review]

**Review by denisbodnar — DISMISSED** [review]

File: server/controllers/event.controller.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+const jwt = require('jsonwebtoken');
+const secret = require('./../../config/jwt.secretkey.json');
+const needEventProperties = {
+  attributes: ['name', 'date_event', 'location_name', 'longitude', 'latitude', 'description']
+};
+
+module.exports = {
+  create(req, res) {
+    let token = req.headers['x-access-token'];
+    let decoder;
```
Comment (4asof4ik) [inline server/controllers/event.controller.js]: Why  you check token agen

Comment (4asof4ik) [reply server/controllers/event.controller.js]: Check token in one place

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

File: server/controllers/event.controller.js (hunk 3 above)
Comment (yarvanya) [reply server/controllers/event.controller.js]: I have understood, sorry, I'll correct it))

**Review by yurii-kokhalevych — DISMISSED** [review]

**Review by denisbodnar — DISMISSED** [review]

**Review by yuliannakorol — DISMISSED** [review]

File: server/models/event.model.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    paranoid: false,
+    timestamps: false,
+    classMethods: {
+      associate: models => {
+        Event.hasOne(models.statusEvent, {
+          foreignKey: 'id'
+        });
+      }
+    },
+    hooks: {
```
Comment (4asof4ik) [inline server/models/event.model.js:52]: we will be using this hooks in each model, so we don't need this copy paste from model to model

File: server/models/event.model.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+module.exports = (sequelize, DataTypes) => {
+  const Event = sequelize.define('Event', {
+    name: {
+      type: DataTypes.STRING,
+      allowNull: false
+    },
+    date_event: {
+      type: DataTypes.DATE
```
Comment (4asof4ik) [inline server/models/event.model.js]: required

File: server/controllers/event.controller.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    .catch(error => {
+        return res.status(500).send(error);
+      });
+  },
+
+  destroy(req, res) {
+    Event.findById(req.params.id).then(event => {
+        if (event.owner !== decoded.id || !event) {
+          return res.status(404).send({
+            message: 'Event has not found! Please try again!',
```
Comment (4asof4ik) [inline server/controllers/event.controller.js]: all messages should be in messages file

File: server/controllers/event.controller.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    .then(event => res.status(200).send(event))
+    .catch(error => res.status(400).send(error));
+  },
+
+  retrieve(req, res) {
+    Event.findById(req.params.id).then(event => {
+        if (event.owner !== decoded.id || !event)  {
+          return res.status(400).send({message: 'Event has not found! Try again!'});
+        } else {
+          let needEventProperties = Object.assign({}, {
```
Comment (4asof4ik) [inline server/controllers/event.controller.js]: find a better way

File: server/controllers/event.controller.js [hunk 8]
```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+const User = require('../../config/db').User;
+const Event = require('../../config/db').Event;
+const jwt = require('jsonwebtoken');
+const secret = require('./../../config/jwt.secretkey.json');
+
+module.exports = {
+  create(req, res) {
+    let assignEvent = Object.assign({}, req.body, {owner: decoded.id});
```
Comment (4asof4ik) [inline server/controllers/event.controller.js]: get owner id from request params

File: server/controllers/event.controller.js [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+      attributes: ['name', 'date_event', 'location_name', 'longitude', 'latitude', 'description'],
+      where: {owner: decoded.id}
+    })
+    .then(event => res.status(200).send(event))
+    .catch(error => res.status(400).send(error));
+  },
+
+  retrieve(req, res) {
+    Event.findById(req.params.id).then(event => {
+        if (event.owner !== decoded.id || !event)  {
```
Comment (4asof4ik) [inline server/controllers/event.controller.js]: req.params.id (owner id)
decoded.id (owner id) ??

File: server/controllers/event.controller.js [hunk 10]
```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+const User = require('../../config/db').User;
+const Event = require('../../config/db').Event;
+const jwt = require('jsonwebtoken');
+const secret = require('./../../config/jwt.secretkey.json');
```
Comment (4asof4ik) [inline server/controllers/event.controller.js]: 5-6 added but not used

File: server/controllers/event.controller.js [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+
+    Event.create(assignEvent)
+      .then(event => res.status(201).send(event))
+      .catch(error => res.status(400).send(error));
+  },
+
+  list(req, res) {
+    Event.findAll({
+      attributes: ['name', 'date_event', 'location_name', 'longitude', 'latitude', 'description'],
+      where: {owner: decoded.id}
```
Comment (4asof4ik) [inline server/controllers/event.controller.js]: check all decoded

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

File: server/controllers/event.controller.js (hunk 8 above)
Comment (yarvanya) [reply server/controllers/event.controller.js]: Here it is impossible to do it.

File: server/models/event.model.js (hunk 4 above)
Comment (yarvanya) [reply server/models/event.model.js:52]: It will be another task for all back-end project, because, now, we have its hooks on dev branch.

**Review by denisbodnar — DISMISSED** [review]

**Review by yuliannakorol — DISMISSED** [review]

**Review by denisbodnar — APPROVED** [review]

**Review by yurii-kokhalevych — APPROVED** [review]

**Review by yuliannakorol — APPROVED** [review]

**Pull request merged**